src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -889,10 +910,36 @@ void AndroidDeviceControllerWrapper::OnICDRegistrationInfoRequired()
 void AndroidDeviceControllerWrapper::OnICDRegistrationComplete(chip::NodeId icdNodeId, uint32_t icdCounter)
 {
     chip::DeviceLayer::StackUnlock unlock;
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::app::ICDClientInfo clientInfo;
+    clientInfo.peer_node         = ScopedNodeId(icdNodeId, Controller()->GetFabricIndex());
+    clientInfo.monitored_subject = mAutoCommissioner.GetCommissioningParameters().GetICDMonitoredSubject().Value();
+    clientInfo.start_icd_counter = icdCounter;
+
+    err = mICDClientStorage.SetKey(clientInfo, mAutoCommissioner.GetCommissioningParameters().GetICDSymmetricKey().Value());
+    if (err == CHIP_NO_ERROR)
```

Comment:
if failed, print error, then early return? currently you are calling onICDRegistrationComplete even thought it fails.

Reply:
The current DevicePairingDelegate implementation does not consider stopping it. Even in other error cases, the error code is transmitted only to the upper layer.

Reply:
if fails, why we continue? maybe it is existing bug?

Reply:
When a failure occurs, commissioning cannot be stopped, so there is concern as to whether it is appropriate to return simply by failing to save. Wouldn't it be more appropriate for a callback function to be called from the Upper Layer when only saving fails?

Reply:
ack